### ISSUE TITLE
[Bug]: Clicking the Accounts list has a serious delay in response.

### ISSUE THREAD

**Issue by cyhhao** [MetaMask/metamask-extension#23913]
### Describe the bug

When I click Accounts at the top, it takes a long time for the Accounts List to appear. I have 28 accounts, which is not too many.

I used Chrome's performance analysis and found the following problems:

### Screenshots/Recordings

![image](https://github.com/MetaMask/metamask-extension/assets/5381613/6c5e8817-0f20-443a-b00f-389af28e4aad)

### Steps to reproduce

0. Own some Accounts and add some Networks
1. Click Metamask and the home page will appear.
2. Click the Accounts toggle button at the top

### Error messages or log output

Based on the performance analysis above, I guess the problem lies in the `getAddressConnectedSubjectMap` function which is very time consuming and can be analyzed using this function as the entry point.

After tracing it, I found that the root cause seemed to be the `unstable_runWithPriority` function.

But I am not familiar with the Metamask code, and further debugging is difficult. I hope to get your help. 

Thanks

### Version

11.13.1

### Build type

None

### Browser

Chrome

### Operating system

MacOS

**Comment by cyhhao** [comment]
After looking at part of the code and conducting tests, I believe I have identified the root cause of the problem: it is related to the large Map of connected sites because I have connected to many websites since using Metamask in 2019.

When I disconnected from each site one by one, the problem was resolved. I think this issue should be addressed as follows:

1. Add a "disconnect all sites" button on the connected sites page to allow users to easily clear this Map with one click.

2. The Map should not be repeatedly read every time an account is switched.

Additionally, I also found that when switching accounts, there are frequent requests for every connected site's icon, which is unnecessary. Therefore:

3. Avoid reading all "connected sites" whenever possible (I can't think of any scenario other than the list page where it's necessary to read all).
